internal/transport/http2_client.go

```diff
@@ -87,8 +87,9 @@ type http2Client struct {
 	// goAway is closed to notify the upper layer (i.e., addrConn.transportMonitor)
 	// that the server sent GoAway on this transport.
 	goAway chan struct{}
-
-	framer *framer
+	// This channel is closed when the keepAlive goroutine exits.
+	keepAliveDone chan struct{}
```

Comment:
Nit: maybe a trialing comment is sufficient for this one.

```
keepAliveDone chan struct{} // Closed when the keepalive goroutine exits.
```

Also s/keepAliveDone/keepaliveDone. 
Keepalive is a single word in our context.

Reply:
Done.